web/src/HUD.tsx

```diff
@@ -259,19 +260,21 @@ export default class HUD extends Component<HudProps, HudState> {
               <LogStoreProvider value={this.state.logStore || new LogStore()}>
                 <ResourceGroupsContextProvider>
                   <ResourceListOptionsProvider>
-                    <Switch>
-                      <Route
-                        path={this.path("/r/:name/overview")}
-                        render={(props: RouteComponentProps<any>) => (
-                          <OverviewResourcePane view={this.state.view} />
-                        )}
-                      />
-                      <Route
-                        render={() => (
-                          <OverviewTablePane view={this.state.view} />
-                        )}
-                      />
-                    </Switch>
+                    <ResourceSelectionProvider>
```

Comment:
i feel like at some point we should look into how we write + organize these providers so that we don't have unnecessary re-renders (for example, there's not really any reason this needs to be re-rendered on every overview render). but this is fine for now

Reply:
totally, that's fair!